stactools_sentinel1/requirements.txt

```diff
@@ -0,0 +1 @@
+rasterio
```

Comment:
IMO it's clearest if each package defines as dependencies all modules that it explicitly imports. It makes it more complicated if you have to remember "well we import `rasterio`, but use it as a side effect of `stactools-core`, so if we ever change that dependency in `stactools-core`, we'll have to remember it has changes here"

Reply:
By that logic shouldn't you also be including `click`, `pystac`, `shapely`, etc?

Reply:
I don't know how this `requirements.txt` is used. But yes, I would argue every external module that is explicitly imported should be defined in `install_requires` in `setup.py`. So if you use `import shapely` in this module, then it should be included in `install_requires`.

Reply:
While I agree that explicit is usually better, I think I disagree with you in this case, because

- DRY
- These packages are tightly coupled such that **stactools_sentinel1** is never going to live in an environment w/o **stactools_core**.
- It's been this repo's convention to [pin](https://github.com/stac-utils/stactools/blob/master/stactools_core/requirements.txt#L7) [dependencies](https://github.com/stac-utils/stactools/blob/master/stactools_planet/requirements.txt#L2) to a tiny version. By repeating the dependency multiple places in the same environment, you might inadvertently overwrite a pinned dependency, e.g. the bare **rasterio** dependency in this PR could do a major/minor rasterio bump over the pinned version in **stactools_core**.

`requirements.txt` usage is inconsistent throughout this repo, so whichever way we break, we should ensure all packages are updated to follow (and we should document it under https://github.com/stac-utils/stactools#adding-a-new-sub-package).

Reply:
Good discussion, in an area of stactools that I think could use a lot of improvement. Some notes:
- I've been in the practice of pinning to specific versions after getting burned by debugging things that are caused by a version upgrade that I didn't expect in a dependency. However, pinning to a specific version in something that is pulled in as a dependency can be a pain if the downstream user's project depends on another version that would work just fine but makes pip unhappy due to version mismatch. I'm not sure the best solution for this - perhaps relaxing the version restrictions to major versions and hoping dependencies keep close to semver?
- There are some empty requirement.txt's around because the replicated setup.py between the datasets automatically looks for them, and it made writing the development environment scripts a bit easier. This isn't great and could change to something better.

I'm not sure I have a good opinion on the 'subpackages be explicit about dependencies' vs 'keep core dependencies in core' discussion; you both are making good points :-)

I'll point out too that the way setup.py is setup doesn't work with building sdist's as described in https://github.com/microsoft/planetary-computer-sdk-for-python/pull/6 - is there a better way to do this sort of subpackage requirements management than using the `-r requirements.txt` method?

Reply:
As this discussion has expanded, it probably shouldn't block this PR, so I've opened #95 to capture this question moving forward.

While trying to avoid wading _fully_ into the "Python packaging is a mess" realm of things :-P, I'd say that I generally agree w/ @TomAugspurger that a setup.cfg buildout is simple and forward-looking. I think it's right that we wouldn't be able to use setup.cfg at the top level (https://github.com/microsoft/planetary-computer-sdk-for-python/pull/6#issuecomment-825674513), but I think we could use setup.cfg each subpackage (since the subpackages are pretty simple to build as a rule) and have the top level be the only place with the fancy setup.py.